kafka/adminapi.go

```diff
@@ -271,6 +299,66 @@ type DescribeConsumerGroupsResult struct {
 	ConsumerGroupDescriptions []ConsumerGroupDescription
 }
 
+// TopicCollection represents a collection of topics.
+type TopicCollection struct {
+	// Slice of topic names.
+	topicNames []string
+}
+
+// NewTopicCollectionOfTopicNames creates a new TopicCollection based on a list
+// of topic names.
+func NewTopicCollectionOfTopicNames(names []string) TopicCollection {
+	return TopicCollection{
+		topicNames: names,
+	}
+}
+
+// Topic Partition information
+type TopicPartitionInfo struct {
+	// Partition id.
+	Partition int
+	// Leader broker.
+	Leader *Node
+	// Replicas of partition.
+	Replicas []Node
+	// In-Sync-Replicas of partition.
+	Isr []Node
```

Comment:
Shouldn't all be pointers or all be concrete type? I know that leader can be null and hence pointer but just a thought for discussion.

Reply:
I like it specifically different because it indicates that Leader can be nullable, but others cannot be nullable. The list can be empty for replicas or isr, but nil cannot be contained in the list.